op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -633,3 +552,44 @@ func timer(name string) func() {
 		log.Info("TIMER", "process", name, "duration", time.Since(start))
 	}
 }
+
+// checkRange loads a range of blocks from the db, it transparently handles loading from the freezer and the normal db.
+func checkRange(freezer *rawdb.Freezer, db ethdb.Database, start, end uint64) error {
+	numAncients, err := freezer.Ancients()
+	if err != nil {
+		return fmt.Errorf("failed to get number of ancients in freezer: %w", err)
+	}
+	var ancientRange, nonAncientRange *RLPBlockRange
+	if start < numAncients {
+		ancientRange, err = loadAncientRange(freezer, start, min(numAncients-1, end)-start+1)
+		if err != nil {
+			return err
+		}
+		if err := ancientRange.CheckContinuity(); err != nil {
+			return err
+		}
+	}
+	if end > numAncients {
```

Comment:
Shouldn't this be >= ?

Reply:
It looks like `end` is inclusive, and `numAncients` would be the number of the first non-ancient block. So if `checkRange` is called with `end == numAncients` we won't read the first non-ancient block

Reply:
Well spotted! Yes